crates/ruff_server/src/server.rs

```diff
@@ -60,6 +61,42 @@ impl Server {
 
         conn.initialize_finish(id, initialize_data)?;
 
+        // Register capabilities
+        conn.sender
+            .send(lsp_server::Message::Request(lsp_server::Request {
+                id: 1.into(),
+                method: "client/registerCapability".into(),
+                params: serde_json::to_value(lsp_types::RegistrationParams {
+                    registrations: vec![lsp_types::Registration {
+                        id: "ruff-server-watch".into(),
+                        method: "workspace/didChangeWatchedFiles".into(),
```

Comment:
Do we need to check if the client supports dynamic `didChangeWatchedFiles` requests https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#didChangeWatchedFilesClientCapabilities

Reply:
We do, good catch.